javagen/src/main/java/com/azure/autorest/template/EnumTemplate.java

```diff
@@ -28,109 +29,127 @@ public static EnumTemplate getInstance() {
     }
 
     public final void write(EnumType enumType, JavaFile javaFile) {
-        String enumTypeComment = String.format("Defines values for %1$s.", enumType.getName());
+        JavaSettings settings = JavaSettings.getInstance();
+        String enumTypeComment = "Defines values for " + enumType.getName() + ".";
         if (enumType.getExpandable()) {
-            javaFile.declareImport("java.util.Collection", "com.fasterxml.jackson.annotation.JsonCreator", getStringEnumImport());
-            javaFile.javadocComment(comment ->
-            {
-                comment.description(enumTypeComment);
+            writeExpandableStringEnum(enumType, javaFile, enumTypeComment, settings);
+        } else {
+            writeEnum(enumType, javaFile, enumTypeComment, settings);
+        }
```

Comment:
Minor changes to prevent the `EnumTemplate` from generating Jackson annotations. One minor change for `enum` types to check if the `value` in `fromString` is null and to fast return null